it/v2/connectors/PersistenceConnectorSpec.scala

```diff
@@ -790,7 +791,136 @@ class PersistenceConnectorSpec
+
+        whenReady(persistenceConnector.postArrival(eoriNumber, source)) {
+          result =>
+            result mustBe okResult
+        }
+    }
+
+    "On an upstream internal server error, get a UpstreamErrorResponse" in forAll(arbitrary[EORINumber]) {
+      eoriNumber =>
+        server.stubFor(
+          post(
+            urlEqualTo(targetUrl(eoriNumber))
+          )
+            .withHeader(HeaderNames.CONTENT_TYPE, equalTo(MimeTypes.XML))
+            .willReturn(
+              aResponse()
+                .withStatus(INTERNAL_SERVER_ERROR)
+                .withBody(
+                  Json.stringify(Json.toJson(PresentationError.internalServiceError()))
+                )
+            )
+        )
+
+        implicit val hc: HeaderCarrier = HeaderCarrier(extraHeaders = Seq(HeaderNames.ACCEPT -> ContentTypes.JSON))
+
+        val source = Source.single(ByteString("<test></test>", StandardCharsets.UTF_8))
+
+        val future = persistenceConnector.postArrival(eoriNumber, source).map(Right(_)).recover {
+          case NonFatal(e) => Left(e)
+        }
+
+        whenReady(future) {
+          result =>
+            result.left.get mustBe a[UpstreamErrorResponse]
+            val response = result.left.get.asInstanceOf[UpstreamErrorResponse]
+            response.statusCode mustBe INTERNAL_SERVER_ERROR
+            Json.parse(response.message).validate[StandardError] mustBe JsSuccess(StandardError("Internal server error", ErrorCode.InternalServerError))
+        }
+    }
+
+    "On an upstream bad request, get an UpstreamErrorResponse" in forAll(arbitrary[EORINumber]) {
+      eoriNumber =>
+        server.stubFor(
+          post(
+            urlEqualTo(targetUrl(eoriNumber))
+          )
+            .withHeader(HeaderNames.CONTENT_TYPE, equalTo(MimeTypes.XML))
+            .willReturn(
+              aResponse()
+                .withStatus(BAD_REQUEST)
+                .withBody(
+                  Json.stringify(Json.toJson(PresentationError.badRequestError("Bad request")))
+                )
+            )
+        )
+
+        implicit val hc: HeaderCarrier = HeaderCarrier(extraHeaders = Seq(HeaderNames.ACCEPT -> ContentTypes.JSON))
+
+        val source = Source.single(ByteString("<test></test>", StandardCharsets.UTF_8))
+
+        val future = persistenceConnector.postArrival(eoriNumber, source).map(Right(_)).recover {
+          case NonFatal(e) => Left(e)
+        }
+
+        whenReady(future) {
+          result =>
+            result.left.get mustBe a[UpstreamErrorResponse]
+            val response = result.left.get.asInstanceOf[UpstreamErrorResponse]
+            response.statusCode mustBe BAD_REQUEST
+            Json.parse(response.message).validate[StandardError] mustBe JsSuccess(StandardError("Bad request", ErrorCode.BadRequest))
+        }
+    }
+
+    "On an incorrect Json fragment, must return a JsResult.Exception" in forAll(arbitrary[EORINumber]) {
```

Comment:
...but it's returning a JsonParseException (though I remember writing this for departures, did I get it wrong there?)

Could we update the text to say:

"On an incorrect Json fragment from transit-moments, must return a JsonParseException"?

Just makes it more explicit!

Reply:
done the changes